### DISCUSSION TITLE
Linking Manifold lib to Microsoft Visual Studio C#

### DISCUSSION THREAD

**Discussion by edumarzolla** [elalish/manifold#870] [Q&A] [answered]
Could someone please explain how to link Manifold in a Winform application developed in C#? 
Thanks,

Example of a desired C# function:

```
        private void button1_Click(global::System.Object sender, global::System.EventArgs e)
        {
            //Create a simple cylinder1
            long cylinder1 = cylinder(10f, 5f, 5f, 360, true);

            //Create a simple cylinder2
            long cylinder2 = cylinder(20f, 1f, 1f, 360, true);

            //Subtract cylinder2 from cylinder1
            long result = boolean_subtract(cylinder1, cylinder2);

            //Export mesh
            int i = ExportMeshFile(result, @"C:\Temp\result.glb");

        }
```

**Comment by elalish** [comment]
@goswinr Did you get this working? Perhaps we should add some documentation on this, especially since none us core contributors are Windows/C# users.

**Reply by goswinr** [reply]
@elalish no success yet. I think it is just a matter of compiling with the Visual C++ compiler. Would be amazing to have windows bindings. I am in the construction industry. Everyone is using windows and most building modeling tools (like Revit or Rhino) are automated via .NET

**Comment by pca006132** [comment]
If you follow the steps in https://stackoverflow.com/questions/11425202/is-it-possible-to-call-a-c-function-from-c-net, do you have any issue? If yes, show us the error messages?

**Comment by weianweigan** [comment, marked as answer]
@edumarzolla  I released a [nuget package](https://www.nuget.org/packages/ManifoldNET.MeshIO/1.0.4-alpha) may be helpful.
```c#
using ManifoldNET;

var cylinder1 = Manifold.Cylinder(10f, 5f, 5f, 360, true);
var cylinder2 = Manifold.Cylinder(20f, 1f, 1f, 360, true);

var result = Manifold.BooleanOperation(cylinder1, cylinder2, BoolOperationType.Subtract);
result.MeshGL.ExportMeshGL("result.glb");
```

I tested it in notebook and it works. 

![image](https://github.com/user-attachments/assets/4b6360b5-5fa7-449e-a3a5-3e75766c81eb)
